src/database/core/view/QueryParams.ts

```diff
@@ -273,25 +260,25 @@ export class QueryParams {
       newParams.indexStartName_ = '';
     }
     return newParams;
-  };
+  }
 
   /**
    * @param {*} indexValue
    * @param {?string=} key
    * @return {!QueryParams}
    */
-  endAt(indexValue, key) {
-    var newParams = this.copy_();
+  endAt(indexValue: any, key?: string | null): QueryParams {
+    const newParams = this.copy_();
     newParams.endSet_ = true;
     if (!(indexValue !== undefined)) {
       indexValue = null;
     }
     newParams.indexEndValue_ = indexValue;
-    if ((key !== undefined)) {
+    if (key !== undefined) {
       newParams.endNameSet_ = true;
       newParams.indexEndName_ = key;
     } else {
-      newParams.startEndSet_ = false;
```

Comment:
`startEndSet_` doesn't exists in this class. I'm 99.9% sure this was meant to say `endNameSet_`, but double-check just in case.